src/legacy/ui/public/visualize/loader/pipeline_data_loader.ts

```diff
@@ -21,9 +21,17 @@ import { RequestHandlerParams, Vis } from '../../vis';
 import { buildPipeline, runPipeline } from './pipeline_helpers';
 
 export class PipelineDataLoader {
+  private abortController?: AbortController;
+
   constructor(private readonly vis: Vis) {}
 
+  public cancel() {
```

Comment:
this one can get abort controller passed in (its internal implementation detail of visualize loader)

Reply:
lookin into this again, if we move this to embedded_visualzie_handler and just pass abortSignal in to data loaders we don't need this duplicated in two places.